src/dmd/backend/cgcse.d

```diff
@@ -0,0 +1,121 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Manage the memory allocated on the runtime stack to save Common Subexpressions (CSE).
+ *
+ * Copyright:   Copyright (C) 1985-1998 by Symantec
+ *              Copyright (C) 2000-2019 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/backend/cgcse.d, backend/cgcse.d)
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/backend/cgcse.d
+ */
+
+module dmd.backend.cgcse;
+
+version (SCPP)
+    version = COMPILE;
+version (MARS)
+    version = COMPILE;
+
+version (COMPILE)
+{
+
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+
+import dmd.backend.cc;
+import dmd.backend.cdef;
+import dmd.backend.code;
+import dmd.backend.code_x86;
+import dmd.backend.el;
+import dmd.backend.global;
+import dmd.backend.ty;
+
+import dmd.backend.barray;
+
+extern (C++):
+nothrow:
+
+__gshared Barray!CSE csextab;     /* CSE table (allocated for each function) */
+
+/****************************
+ * Table of common subexpressions stored on the stack.
+ *      csextab[]       array of info on saved CSEs
+ *      CSEpe           pointer to saved elem
+ *      CSEregm         mask of register that was saved (so for multi-
+ *                      register variables we know which part we have)
+ */
+
+enum CSEload       = 1;       // set if the CSE was ever loaded
```

Comment:
Public declaration so DDOC style slashes comments. This comment counts also for pretty much everything that's commented in the module (unless you meant to set `private`  CSE fields)

Reply:
For easy review, I kept this code movement as cut/paste. It's really hard to review such if it is all refactored as well as being moved. As I explain again and again, small incremental PRs are far easier to review than large ones, and far easier to debug if a regression is introduced. In light of that, this should be reviewed as progress towards a goal, not whether the goal is achieved.